src/meshio/netgen/_netgen.py

```diff
@@ -0,0 +1,361 @@
+            "cd2names",
+            "cd3names",
+            "face_colours",
+            "identifications",
+            "materials",
+            "singular_edge_left",
+            "singular_edge_right",
+            "singular_face_inside",
+            "singular_face_outside",
+            "singular_points",
+        ]:
+            _skip_block(f)
+        else:
+            raise RuntimeError(f"Unknown Netgen mesh token: {line}")
+
+    # convert to numpy arrays
+    # subtract one (netgen counts 1-based)
+    # apply permutation of vertex numbers
+    for k, (t, data) in enumerate(cells):
+        pmap = netgen_to_meshio_pmap[t]
+        d = np.array(data, dtype=np.uint32)
+        d[:, :] = d[:, pmap] - 1
+        cells[k] = (t, d)
+
+    mesh = Mesh(points, cells, cell_data={"netgen:index": cells_index})
+    return mesh
+
+
+def write(filename, mesh, float_fmt=".16e"):
+    if str(filename).endswith(".vol.gz"):
+        import gzip
+
+        with gzip.open(filename, "wt") as f:
+            return write_buffer(f, mesh, float_fmt)
+
+    with open_file(filename, "w") as f:
+        return write_buffer(f, mesh, float_fmt)
+
+
+def write_buffer(f, mesh, float_fmt):
+    num_points, dimension = mesh.points.shape
+    cells_per_dim = [0, 0, 0, 0]
+    cells_index = (
+        mesh.cell_data["netgen:index"]
+        if "netgen:index" in mesh.cell_data
+        else [None] * len(mesh.cells)
+    )
+    for block in mesh.cells:
+        cells_per_dim[_topological_dimension[block.type]] += len(block)
+
+    f.write(f"# Generated by meshio {__version__}\n")
+    f.write("mesh3d\n\n")
+
+    f.write("dimension\n")
+    f.write(f"{dimension}\n\n")
+
+    f.write("geomtype\n")
+    f.write("0\n")
+
+    f.write("\n# surfnr    bcnr   domin  domout      np      p1      p2      p3\n")
+    f.write("surfaceelements\n")
+    f.write(f"{cells_per_dim[2]}\n")
+    for block, index in zip(mesh.cells, cells_index):
+        if _topological_dimension[block.type] == 2:
+            _write_cells(f, block, index)
+
+    f.write("\n#  matnr      np      p1      p2      p3      p4\n")
+    f.write("volumeelements\n")
+    f.write(f"{cells_per_dim[3]}\n")
+    for block, index in zip(mesh.cells, cells_index):
+        if _topological_dimension[block.type] == 3:
+            _write_cells(f, block, index)
+
+    f.write(
+        "\n# surfid  0   p1   p2   trignum1    trignum2   domin/surfnr1    domout/surfnr2   ednr1   dist1   ednr2   dist2\n",
+    )
+    f.write("edgesegmentsgi2\n")
+    f.write(f"{cells_per_dim[1]}\n")
+    for block, index in zip(mesh.cells, cells_index):
+        if _topological_dimension[block.type] == 1:
+            _write_cells(f, block, index)
+
+    f.write("\n#          X             Y             Z\n")
+    f.write("points\n")
+    f.write(f"{len(mesh.points)}\n")
+
+    points = mesh.points
+    if dimension == 2:
+        p = np.zeros((points.shape[0], 3), dtype=points.dtype)
+        p[:, :2] = points
+    else:
```

Comment:
Let's skip the `else` here and just use `points`. The `if` condition can use `np.column_stack`